packages/esm-appointments-app/src/hooks/useAppointmentList.tsx

```diff
@@ -13,48 +14,61 @@ interface AppointmentPatientList {
     name: string;
     uuid: string;
     age: number;
-    identifiers: Array<{ identifierName: string; identifier: string }>;
+    identifier: string;
   };
   providers: Array<Provider>;
   service: AppointmentService;
   startDateTime: string;
 }
 
-export const useAppointmentList = (appointmentStatus: string, startDate?: string, identifierType?: string) => {
+export const useAppointmentList = (appointmentStatus: string, date?: string) => {
   const { currentAppointmentDate } = useAppointmentDate();
-  const forDate = startDate ? startDate : currentAppointmentDate;
-  const url = `/ws/rest/v1/appointment/appointmentStatus?status=${appointmentStatus}&forDate=${forDate}`;
+  const startDate = date ? date : currentAppointmentDate;
+  const endDate = dayjs(startDate).endOf('day').format('YYYY-MM-DDTHH:mm:ss.SSSZZ'); // TODO: fix? is this correct?
+  const searchUrl = `/ws/rest/v1/appointments/search`;
+  const abortController = new AbortController();
+
+  const fetcher = ([url, startDate, endDate, status]) =>
+    openmrsFetch(url, {
+      method: 'POST',
+      signal: abortController.signal,
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: {
+        startDate: startDate,
+        endDate: endDate,
+        status: status,
+      },
+    });
 
   const { data, error, isLoading } = useSWR<{ data: Array<AppointmentPatientList> }>(
-    appointmentStatus ? url : null,
-    openmrsFetch,
+    [searchUrl, startDate, endDate, appointmentStatus],
+    fetcher,
     { errorRetryCount: 2 },
   );
 
-  const appointments = data?.data?.map((appointment) => toAppointmentObject(appointment, identifierType));
+  const appointments = data?.data?.map((appointment) => toAppointmentObject(appointment));
   return { appointmentList: (appointments as Array<any>) ?? [], isLoading, error };
 };
 
-export const useEarlyAppointmentList = (startDate?: string, identifierType?: string) => {
+export const useEarlyAppointmentList = (startDate?: string) => {
   const { currentAppointmentDate } = useAppointmentDate();
   const forDate = startDate ? startDate : currentAppointmentDate;
   const url = `/ws/rest/v1/appointment/earlyAppointment?forDate=${forDate}`;
 
   const { data, error, isLoading } = useSWR<{ data: Array<AppointmentPatientList> }>(url, openmrsFetch, {
     errorRetryCount: 2,
   });
-  const appointments = data?.data?.map((appointment) => toAppointmentObject(appointment, identifierType));
+  const appointments = data?.data?.map((appointment) => toAppointmentObject(appointment));
   return { earlyAppointmentList: (appointments as Array<any>) ?? [], isLoading, error };
 };
 
-function toAppointmentObject(appointment: AppointmentPatientList, identifierType: string) {
-  const patientIdentifier = appointment.patient.identifiers.find(
-    (identifier) => identifier.identifierName === identifierType,
```

Comment:
One key change here.... the Bahmni Appointments endpoint just returns a single identifier (the preferred patient identifier) while the custom endpoint returns all the identifiers and allows the identifier to display to be configured.  I've removed this functionality, we should discuss @CynthiaKamau  what the KenyaEMR needs here and how best to support it.

Reply:
Hmm... it looks like there have been some recent changes to the endpoint to return all active identifiers... we can likely use this to pull out the proper identifier. I will look into ticketing this.

Reply:
Pulling the active identifier will work for us

Reply:
Great!